pallets/capacity/src/lib.rs

```diff
@@ -299,4 +346,41 @@ impl<T: Config> Pallet<T> {
 	fn calculate_capacity(amount: BalanceOf<T>) -> BalanceOf<T> {
 		amount
 	}
+
+	/// Decrease a staking account's active token and create an unlocking chunk to be thawed at some future block.
+	fn decrease_active_staking_balance(unstaker: &T::AccountId, amount: BalanceOf<T>) -> DispatchResult {
+		let mut staking_account_details = Self::get_staking_account_for(unstaker).ok_or(Error::<T>::NotStakingAccount)?;
+		ensure!(amount <= staking_account_details.active, Error::<T>::AmountToUnstakeExceedsAmountStaked);
+
+		let current_block: T::BlockNumber = frame_system::Pallet::<T>::block_number();
+		let thaw_at = current_block.saturating_add(T::BlockNumber::from(T::UnstakingThawPeriod::get()));
+
+		staking_account_details.decrease_by(amount, thaw_at)?;
+		Self::set_staking_account(&unstaker, &staking_account_details);
+
+		Ok(())
+	}
+
+	/// Reduce available capacity of target and return the amount of capacity reduction.
+	fn reduce_capacity(unstaker: &T::AccountId, target: MessageSourceId, amount: BalanceOf<T>) -> Result<BalanceOf<T>, DispatchError>  {
+		let mut staking_target_details = Self::get_target_for(&unstaker, &target).ok_or(Error::<T>::NotStakingAccount)?;
```

Comment:
I think we want a different error here: NotStakingAccount
maybe we can use `InvalidTarget`.

Reply:
In this case, `InvalidTarget` might not be a good error. Maybe we reserve the `InvalidTarget` error when a target does not exist. In this case we are throwing an error because the Staker did not target the provider to receive Capacity. Maybe we need another one like `StakerTargetInvalid.`

Reply:
@shannonwells thoughts?

Reply:
This is one of those cases where the storage is in an invalid state. We _shouldn't_ ever reach a state where we have a staking account storage but not a staking target. In addition to a separate error for this case, we probably want to know if it ever happens and trigger a log in our logging infrastructure

Reply:
I don't think we need to log it since the error will be included in the block.